File: 2022/Day2/solution.py
def part1(_input: str) -> int:
    letter2value = {"A":0, "B":1, "C":2, "X":0, "Y":1, "Z":2}
    with open(_input, "r", encoding="UTF-8") as file:
        return sum(letter2value[game[2]] + 1 + ((letter2value[game[2]] - letter2value[game[0]] + 1) % 3) * 3 for game in file.readlines())

def part2(_input: str) -> int:
    letter2value = {"A":0, "B":1, "C":2, "X":0, "Y":1, "Z":2}
    with open(_input, "r", encoding="UTF-8") as file:
        return sum(letter2value[game[2]] * 3 + ((letter2value[game[0]] + letter2value[game[2]] - 1) % 3) + 1 for game in file.readlines())

_input = "./2022/Day2/input.txt"

print("Part 1:", part1(_input))
print("Part 2:", part2(_input))

File: 2022/Day2/solution.c
#include <stdio.h>

#define B 'X'
#define A 'A'

int mod(int a, int b) {
	int r;
	r = a % b;
	return r < 0? r + b : r;
}

int part1() {
	int total = 0;
	char a, b;
	while (scanf("%c %c\n", &a, &b) != EOF) {
		total += b - B + 1 + mod(b - a + A - B + 1, 3) * 3;
	}
	printf("Part 1: %d\n", total);
	return 0;
}

int part2() {
	int total = 0;
	char a, b;
	while (scanf("%c %c\n", &a, &b) != EOF) {
		total += (b - B) * 3 + 1 + mod(a + b - A - B - 1, 3);
	}
	printf("Part 2: %d\n", total);
	return 0;
}

int main() {
	//part1();
	part2();
	return 0;
}
